zerver/lib/notifications.py

```diff
@@ -269,7 +281,7 @@ def do_send_missedmessage_events_reply_in_zulip(user_profile, missed_messages, m
 
     subject = "Missed Zulip%s from %s" % (plural_messages, sender_str)
     from_email = 'Zulip <%s>' % (settings.NOREPLY_EMAIL_ADDRESS,)
-    if len(senders) == 1 and settings.SEND_MISSED_MESSAGE_EMAILS_AS_USER:
+    if len(senders) == 1 and settings.SEND_MISSED_MESSAGE_EMAILS_AS_USER and settings.EMAIL_GATEWAY_PATTERN:
```

Comment:
I think you don't want this -- this is for making replies go to the sender via email, not Zulip.